src/app/cli/src/commands/export_command.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use kamu::domain::{ExportFormat, ExportService, QueryService};
+use opendatafabric::DatasetRef;
+
+use crate::{CLIError, Command};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct ExportCommand {
+    export_service: Arc<dyn ExportService>,
+    query_service: Arc<dyn QueryService>,
+    dataset_ref: DatasetRef,
+    output_path: Option<PathBuf>,
+    output_format: String,
+    partition_size: Option<usize>,
+    quiet: bool,
+}
+
+impl ExportCommand {
+    pub fn new(
+        export_service: Arc<dyn ExportService>,
+        query_service: Arc<dyn QueryService>,
+        dataset_ref: DatasetRef,
+        output_path: Option<PathBuf>,
+        output_format: String,
+        partition_size: Option<usize>,
+        quiet: bool,
+    ) -> Self {
+        Self {
+            export_service,
+            query_service,
+            dataset_ref,
+            output_path,
+            output_format,
+            partition_size,
+            quiet,
+        }
+    }
+
+    fn parse_format(&self) -> Result<ExportFormat, CLIError> {
+        match self.output_format.as_str() {
+            "parquet" => Ok(ExportFormat::Parquet),
+            "ndjson" => Ok(ExportFormat::NdJson),
+            "csv" => Ok(ExportFormat::Csv),
+            _ => Err(CLIError::usage_error(format!(
+                "Invalid output format '{}'. Supported formats: 'parquet', 'ndjson', and 'csv'",
+                &self.output_format
+            ))),
+        }
+    }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Command for ExportCommand {
+    async fn run(&mut self) -> Result<(), CLIError> {
+        let format = self.parse_format()?;
+
+        let df = self
+            .query_service
+            .get_data(&self.dataset_ref)
+            .await
+            .map_err(CLIError::failure)?;
+
+        let mut default_path: PathBuf = PathBuf::new();
+        default_path.push(self.dataset_ref.to_string());
+        default_path.push(""); // ensure trailing slash to have it as a dir
```

Comment:
Maybe something like this:

```   
   let path = env::current_dir()?;
   path.join(self.dataset_ref.to_string())
```   
also, should we add some file extension depending on the format?

Reply:
As I understand, the extensions are automatically attached by the `DataFusion` writers. Did you test what happens if I specify particular extensions in the path? Wrong extensions for the selected format?

Reply:
it accepts any extension you specify. you simply need to have any. otherwise it will assume path as a dir

Reply:
here's related datafusion writer https://github.com/apache/datafusion/blob/1fc776981496c0f0648ece8069992c65a7021c83/datafusion/core/src/datasource/file_format/write/demux.rs#L67